gc.c

```diff
@@ -1316,6 +1316,30 @@ tick(void)
     return val;
 }
 
+#elif defined(__POWERPC__) && defined(__APPLE__)
+typedef unsigned long long tick_t;
+#define PRItick "llu"
+
+static __inline__ tick_t
+tick(void)
+{
+  unsigned long long int result=0;
+  unsigned long int upper, lower, tmp;
+  __asm__ volatile(
+                "0:                  \n"
+                "\tmftbu   %0        \n"
+                "\tmftb    %1        \n"
+                "\tmftbu   %2        \n"
+                "\tcmpw    %2,%0     \n"
+                "\tbne     0b        \n"
+                : "=r"(upper),"=r"(lower),"=r"(tmp)
+                );
+  result = upper;
+  result = result<<32;
+  result = result|lower;
+  return(result);
```

Comment:
Would you adjust indents?

Reply:
How about mix more C?
```suggestion
    unsigned long int upper, lower, tmp;
# define mftbu(r) __asm__ volatile("mftbu   %0" : "=r"(r))
# define mftb(r)  __asm__ volatile("mftb    %0" : "=r"(r))
    do {
        mftbu(upper);
        mftb(lower);
        mftbu(tmp);
    } while (tmp != upper);
    return ((tick_t)upper << 32) | lower;
```

Reply:
Should we adopt this change?

Reply:
@nobu @ioquatix Sorry that I missed this. I will try the build with suggested change now.